### PR TITLE
Add Golang JJB templates

### PR DESCRIPTION
**Pull request by JPWKU**
Signed-off-by: Jeremy Phelps <jphelps@linuxfoundation.org>

### CODE REVIEW

**Review by trcox — DISMISSED** [review]

**Review by jpwhitemn — APPROVED** [review]

**Comment by JPWKU** [comment]
@trcox @jpwhitemn I am thinking of doing a top level Makefile.  Will the present a problem for Windows users when building go services?

File: jjb/edgex-templates-go.yaml [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    # yamllint disable-line rule:key-duplicates
+    <<: *go_merge_boiler_plate
+
+
+    builders:
+      - lf-jacoco-nojava-workaround
+      - lf-provide-maven-settings:
+          global-settings-file: 'global-settings'
+          settings-file: '{mvn-settings}'
+      - shell: |
```
Comment (tykeal) [inline jjb/edgex-templates-go.yaml]: Shell scripts should be done as includes and not inlined

File: jjb/edgex-templates-go.yaml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    <<: *go_job_boiler_plate
+    # yamllint disable-line rule:key-duplicates
+    <<: *go_verify_boiler_plate
+
+    builders:
+      - lf-jacoco-nojava-workaround
+      - lf-provide-maven-settings:
+          global-settings-file: 'global-settings'
+          settings-file: '{mvn-settings}'
+      - shell: |
```
Comment (tykeal) [inline jjb/edgex-templates-go.yaml]: Shell scripts should be done as includes and not inlined

File: jjb/edgex-templates-go.yaml [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    # yamllint disable-line rule:key-duplicates
+    <<: *go_merge_boiler_plate
+
+
+    builders:
+      - lf-jacoco-nojava-workaround
+      - lf-provide-maven-settings:
+          global-settings-file: 'global-settings'
+          settings-file: '{mvn-settings}'
+      - shell: |
```
Comment (tykeal) [inline jjb/edgex-templates-go.yaml]: Shell scripts should be done as includes and not inlined

File: jjb/edgex-templates-go.yaml [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    <<: *go_job_boiler_plate
+    # yamllint disable-line rule:key-duplicates
+    <<: *go_verify_boiler_plate
+
+    builders:
+      - lf-jacoco-nojava-workaround
+      - lf-provide-maven-settings:
+          global-settings-file: 'global-settings'
+          settings-file: '{mvn-settings}'
+      - shell: |
```
Comment (tykeal) [inline jjb/edgex-templates-go.yaml]: Shell scripts should be done as includes and not inlined

**Review by tykeal — CHANGES_REQUESTED** [review]
Please break out the shell scripts into includes so that ShellCheck can run across them

**Comment by JPWKU** [comment]
@tykeal the build commands will be contained in Makefile and driven that way.  I'd like to go ahead and get this merged so we can incrementally build it out.

**Review by tykeal — APPROVED** [review]

**Pull request merged**